app/helpers/Intersect.js

```diff
@@ -210,86 +179,150 @@ function binarySearchForTime(array, targetTime) {
-                });
-                SetStoreData(AUTHORITY_NEWS, name_news);
-              });
-
-              // TODO: Look at "publish_date_utc".  We should notify users if
-              //       their authority is no longer functioning.)
-
-              IntersectSet(responseJson.concern_points, dayBin => {
-                if (dayBin !== null && dayBin.reduce((a, b) => a + b, 0) > 0) {
-                  PushNotification.localNotification({
-                    title: languages.t('label.push_at_risk_title'),
-                    message: languages.t('label.push_at_risk_message'),
-                  });
-                }
-              });
-            });
-
-          let nowUTC = new Date().toISOString();
-          let unixtimeUTC = Date.parse(nowUTC);
-          // Last checked key is not being used atm. TODO check this to update periodically instead of every foreground activity
-          SetStoreData(LAST_CHECKED, unixtimeUTC);
-        }
-      } else {
-        console.log('No authority list');
-        return;
+        // TODO: Look at "publish_date_utc".  We should notify users if
+        //       their authority is no longer functioning.)
+
+        // add the current set of points to the set to be intersected later
+        //
+        // TODO:  This currently causes us to generate a single large single array from
+        //    all authorities combined together.  The ideal would be to process in chunks, so
+        //    that no matter how much data was sent by the authority, the memory footprint
+        //    doesn't get too big.
+        //
+        concernResultsCombined = [
+          ...concernResultsCombined,
+          ...responseJson.concern_points,
+        ];
+      } catch (error) {
+        console.log('[authority] fetch/parse error :', error);
       }
-    })
-    .catch(error => console.log('Failed to load authority list', error));
+    }
+  }
+
+  // Set the news for the authorities found.  There's an implicit assumption that news only
+  //  comes from the configured authorities
+  SetStoreData(AUTHORITY_NEWS, name_news);
+
+  // now, get the location data, normalize and do the intersection
+  let locationArrayString = await GetStoreData(LOCATION_DATA);
+  if (locationArrayString !== null) {
+    let locationArray = JSON.parse(locationArrayString);
+    let concernArray = concernResultsCombined;
+    dayBin = intersectSetIntoBins(locationArray, concernArray, dayBin);
+    if (dayBin !== null && dayBin.reduce((a, b) => a + b, 0) > 0) {
+      PushNotification.localNotification({
+        title: languages.t('label.push_at_risk_title'),
+        message: languages.t('label.push_at_risk_message'),
+      });
+    }
+  }
+  console.log('Crossing results: ', dayBin);
+
+  // store the results
+  SetStoreData(CROSSED_PATHS, dayBin); // TODO: Store per authority?
+
+  // the current time is the last time checked
+  let nowUTC = new Date().toISOString();
+  let unixtimeUTC = Date.parse(nowUTC);
```

Comment:
Yeah, much cleaner.

Reply:
JS dates don't maintain the timezoneOffset.. meaning.. if you set a date at now (UTC) it will still have your local getTimezoneOffset

We definitely need tests on that, and I suspect dayjs().utc() might be required to keep things sane, code wise.

Reply:
relevant link that @efalkner sent to me: https://www.youtube.com/watch?v=EUzPpXhFmG8

Takeaway:

> time math is more complicated than we think, use a lib.